controllers/openstackcluster_controller.go

```diff
@@ -431,10 +431,12 @@ func reconcileNetworkComponents(scope scope.Scope, cluster *clusterv1.Cluster, o
 
 	scope.Logger().Info("Reconciling network components")
 
-	err = networkingService.ReconcileExternalNetwork(openStackCluster)
-	if err != nil {
-		handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to reconcile external network: %w", err))
-		return fmt.Errorf("failed to reconcile external network: %w", err)
+	if !openStackCluster.Spec.DisableAPIServerFloatingIP || openStackCluster.Spec.NodeCIDR != "" {
```

Comment:
We will reconcile external network either if APIServerFloatingIP is enabled (*), meaning we need to create a floating IP somewhere, or if NodeCIDR is set, meaning we're going to create the cluster default network.

We also need to do it if there's a Bastion defined.

* I un-doubled the negative!

Reply:
We now have:

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/a44323cc3cb23b021a4eade9c29111a7bff0691e/pkg/cloud/services/networking/network.go#L83-L87

If this is getting too complicated, I suggest we abandon this PR, because we also have the bastion to handle and then all networking related parameters :-/.

Now we have a simple API to not reconcile external networks.